x-pack/plugins/file_upload/public/components/geojson_upload_form/geojson_upload_form.tsx

```diff
@@ -42,35 +39,24 @@ interface State {
   hasFile: boolean;
   isPointsOnly: boolean;
   indexNames: string[];
+  updatedIndexName: string;
 }
 
 export class GeoJsonUploadForm extends Component<Props, State> {
-  private _isMounted = false;
-
   state: State = {
     hasFile: false,
     isPointsOnly: false,
     indexNames: [],
+    updatedIndexName: '',
```

Comment:
Index name State should only be tracked in a single location. Lets remove updatedIndexName from GeoJsonUploadForm state and instead call onIndexNameChange when a file is selected or cleared to propagate the index name change.